sklearn/metrics/_classification.py

```diff
@@ -731,10 +757,13 @@ def jaccard_score(
     sample_weight : array-like of shape (n_samples,), default=None
         Sample weights.
 
-    zero_division : "warn", {0.0, 1.0}, default="warn"
+    zero_division : "warn", 0.0, 1.0, np.nan, default="warn"
         Sets the value to return when there is a zero division, i.e. when there
-        there are no negative values in predictions and labels. If set to
-        "warn", this acts like 0, but a warning is also raised.
+        there are no negative values in predictions and labels.
+
+        Notes:
+        - If set to "warn", this acts like 0, but a warning is also raised.
+        - If set to np.nan, such values will be excluded from the average.
```

Comment:
Do you think this docstring is consistent with the following behavior?

```python
jaccard_score([0, 0, 0], [0, 0, 0], zero_division=np.nan)
# np.nan
```

In this case, everything is excluded and the return is still nan?

Reply:
if everything is excluded there's nothing to compute so the result is undefined, hence np.nan